### PR TITLE
feat(data-warehouse): query tabs

### PR DESCRIPTION
**Pull request by EDsCODE**
## Problem

- want to be able to query multiple things at once side by side

## Changes

- add query tabs

Screenshot 2024-07-25 at 6 32 59 PM

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

### CODE REVIEW

File: frontend/src/lib/monaco/codeEditorLogic.tsx [hunk 1]
```diff
@@ -119,6 +124,60 @@ export const codeEditorLogic = kea<codeEditorLogicType>([
...
+                removeModel: (state, { modelName }) =>
+                    state.filter((model) => model.toString() !== modelName.toString()),
+            },
+        ],
+    }),
+    listeners(({ props, values, actions }) => ({
+        createModel: () => {
+            if (props.monaco) {
+                const uri = props.monaco.Uri.parse((values.modelCount + 1).toString())
+                const model = props.monaco.editor.createModel('SELECT event FROM events LIMIT 100', props.language, uri)
```
Comment (Gilbert09) [inline frontend/src/lib/monaco/codeEditorLogic.tsx:153]: Should we use the default query from `frontend/src/queries/examples.ts`?

File: frontend/src/queries/nodes/HogQLQuery/HogQLQueryEditor.tsx [hunk 2]
```diff
@@ -241,3 +250,36 @@ export function HogQLQueryEditor(props: HogQLQueryEditorProps): JSX.Element {
...
+interface QueryTabProps {
+    model: Uri
+    active?: boolean
+    onClick?: (model: Uri) => void
+    onClear?: (model: Uri) => void
+}
+
+function QueryTab({ model, active, onClear, onClick }: QueryTabProps): JSX.Element {
+    return (
+        <button
```
Comment (Gilbert09) [inline frontend/src/queries/nodes/HogQLQuery/HogQLQueryEditor.tsx:264]: Should we use LemonTabs or something similar for this?

File: frontend/src/queries/nodes/HogQLQuery/HogQLQueryEditor.tsx [hunk 3]
```diff
@@ -241,3 +250,36 @@ export function HogQLQueryEditor(props: HogQLQueryEditorProps): JSX.Element {
...
+            onClick={() => onClick?.(model)}
+            className={clsx(
+                'space-y-px rounded-t p-1 flex flex-row items-center gap-1 hover:bg-[var(--bg-light)] cursor-pointer',
+                active ? 'bg-[var(--bg-light)] border' : 'bg-bg-3000',
+                onClear ? 'pl-3 pr-2' : 'px-4'
+            )}
+        >
+            {'Query ' + model.path.split('/').pop()}
```
Comment (Gilbert09) [inline frontend/src/queries/nodes/HogQLQuery/HogQLQueryEditor.tsx:272]: Will we allow people to name these at some point?

**Review by Gilbert09 — APPROVED** [review]
I think it'd be really good to be able to preserve your tabs and the contents of your tabs within local storage!

File: frontend/src/queries/nodes/HogQLQuery/HogQLQueryEditor.tsx (hunk 2 above)
Comment (EDsCODE) [reply frontend/src/queries/nodes/HogQLQuery/HogQLQueryEditor.tsx:264]: avoided lemontabs as I wanted custom UI that was very much editor-tab like

Comment (EDsCODE) [reply frontend/src/queries/nodes/HogQLQuery/HogQLQueryEditor.tsx:264]: just updated the images in the PR description to match

**Comment by EDsCODE** [comment]
@Gilbert09 one more pass please on the state logic

**Review by Gilbert09 — APPROVED** [review]

**Comment by Twixes** [comment]
Hmm, why is this better than browser tabs? Tabs have come up at times in the past (@corywatilo can tell you a lot about it), but in the browser context, they easily make things more confusing by duplicating browser functionality.

**Comment by EDsCODE** [comment]
@Twixes maybe not the strongest point here but gotten this request internally and externally because it would make flipping between queries part of the posthog experience. I don't think people prefer to have multiple tabs of posthog open. Can dig more here

**Pull request merged**